src/components/navigation/navigation.component.jsx

```diff
@@ -1,31 +1,23 @@
 import React from "react";
+import { connect } from "react-redux";
 // Components
 import NavigationDesktop from "../navigation-desktop/navigation-desktop.component.jsx";
 import NavigationMobile from "../navigation-mobile/navigation-mobile.component.jsx";
+// Actions
+import { setWidth } from "../../redux/navigation/navigation.actions";
 
-//TODO replace this method of handling resizing with a better one - react-sizes library
-class Navigation extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            width: window.innerWidth,
-        };
-        this.handleWindowSizeChange = this.handleWindowSizeChange.bind(this);
-    }
-    UNSAFE_componentWillMount() {
-        window.addEventListener("resize", this.handleWindowSizeChange);
-    }
-    componentWillUnmount() {
-        window.removeEventListener("resize", this.handleWindowSizeChange);
-    }
-    handleWindowSizeChange() {
-        this.setState({ width: window.innerWidth });
-    }
-    render() {
-        const isMobile = this.state.width < 768;
+const Navigation = ({ width, setWidth }) => {
```

Comment:
I don't think that using Redux just to handle screen width and use that information just in one place is a good idea. Using Redux doesn't mean that all the state connected features has to be solved by Redux, It is actually a good practice to find balance and know to when use Redux store and when to use react state. Here when you use that width property in one place there's no point in storing that in global store. Besides on Devtools when resizing redux logger goes crazy :p.

Reply:
True logger goes bananas, width isn't going to be used elsewhere. I've just gone on the redux-spree :p